src/ring_theory/witt_vector/structure_polynomial.lean

```diff
@@ -128,9 +132,132 @@ begin
+  rw [← witt_structure_rat_rec_aux p Φ n, mul_comm, mul_assoc,
+      ← C_mul, mul_one_div_cancel, C_1, mul_one],
+  exact pow_ne_zero _ (nat.cast_ne_zero.2 $ ne_of_gt (nat.prime.pos ‹_›)),
+end
+
+/-- `witt_structure_int Φ` is a family of polynomials `ℕ → mv_polynomial (idx × ℕ) ℚ`
+that are uniquely characterised by the property that
+`bind₁ (witt_structure_int p Φ) (witt_polynomial p ℚ n) = bind₁ (λ i, (rename (prod.mk i) (witt_polynomial p ℚ n))) Φ`.
+In other words: evaluating the `n`-th Witt polynomial on the family `witt_structure_int Φ`
+is the same as evaluation `Φ` on the (appropriately renamed) `n`-th Witt polynomials.
+
+See `witt_structure_int_prop` for this property,
+and `witt_structure_int_exists_unique` for the fact that `witt_structure_int`
+gives the unique family of polynomials with this property. -/
+noncomputable def witt_structure_int (Φ : mv_polynomial idx ℤ) (n : ℕ) : mv_polynomial (idx × ℕ) ℤ :=
+finsupp.map_range rat.num (rat.coe_int_num 0)
+  (witt_structure_rat p (map (int.cast_ring_hom ℚ) Φ) n)
+
+variable {p}
+
+lemma bind₁_rename_expand_witt_polynomial (Φ : mv_polynomial idx ℤ) (n : ℕ)
+  (IH : ∀ m : ℕ, m < (n + 1) →
+    map (int.cast_ring_hom ℚ) (witt_structure_int p Φ m) = witt_structure_rat p (map (int.cast_ring_hom ℚ) Φ) m) :
+  bind₁ (λ b, rename (λ i, (b, i)) (expand p (W_ ℤ n))) Φ =
+  bind₁ (λ i, expand p (witt_structure_int p Φ i)) (W_ ℤ n) :=
+begin
+  apply mv_polynomial.map_injective (int.cast_ring_hom ℚ) int.cast_injective,
+  simp only [map_bind₁, map_rename, map_expand, rename_expand, map_witt_polynomial],
+  have key := (witt_structure_rat_prop p (map (int.cast_ring_hom ℚ) Φ) n).symm,
+  apply_fun expand p at key,
+  simp only [expand_bind₁] at key,
+  rw key, clear key,
+  apply eval₂_hom_congr' rfl _ rfl,
+  rintro i hi -,
+  rw [witt_polynomial_vars, finset.mem_range] at hi,
+  simp only [IH i hi],
+end
+
+lemma C_p_pow_dvd_bind₁_rename_witt_polynomial_sub_sum (Φ : mv_polynomial idx ℤ) (n : ℕ)
+  (IH : ∀ m : ℕ, m < n →
+    map (int.cast_ring_hom ℚ) (witt_structure_int p Φ m) = witt_structure_rat p (map (int.cast_ring_hom ℚ) Φ) m) :
+  C ↑(p ^ n) ∣
+    (bind₁ (λ (b : idx), rename (λ i, (b, i)) (witt_polynomial p ℤ n)) Φ -
+      ∑ i in range n, C (↑p ^ i) * witt_structure_int p Φ i ^ p ^ (n - i)) :=
+begin
+  cases n,
+  { simp only [is_unit_one, int.coe_nat_zero, int.coe_nat_succ, zero_add, pow_zero, C_1, is_unit.dvd] },
+  rw [nat.succ_eq_add_one, C_dvd_iff_zmod, ring_hom.map_sub, sub_eq_zero, map_bind₁],
+  simp only [map_rename, map_witt_polynomial, witt_polynomial_zmod_self],
+  -- prepare a useful equation for rewriting
+  have key := bind₁_rename_expand_witt_polynomial Φ n IH,
+  apply_fun (map (int.cast_ring_hom (zmod (p ^ (n + 1))))) at key,
+  conv_lhs at key { simp only [map_bind₁, map_rename, map_expand, map_witt_polynomial] },
+  rw key,
+  clear key IH,
+  -- clean up and massage
+  rw [bind₁, aeval_witt_polynomial, ring_hom.map_sum, ring_hom.map_sum, finset.sum_congr rfl],
```

Comment:
I'm not so convinced by the order of the steps in this proof: simplify a bit, introduce a new lemma, simplify a bit more. (Not that I can find any way to make this less painful!) How about:

```suggestion

  -- prepare a useful equation for rewriting
  have key := bind₁_rename_expand_witt_polynomial Φ n IH,
  apply_fun (map (int.cast_ring_hom (zmod (p ^ (n + 1))))) at key,
  conv_lhs at key { simp only [map_bind₁, map_rename, map_expand, map_witt_polynomial] },

  -- clean up and massage
  rw [nat.succ_eq_add_one, C_dvd_iff_zmod, ring_hom.map_sub, sub_eq_zero, map_bind₁],
  simp only [map_rename, map_witt_polynomial, witt_polynomial_zmod_self],
  rw key, clear key IH,
  rw [bind₁, aeval_witt_polynomial, ring_hom.map_sum, ring_hom.map_sum, finset.sum_congr rfl],
```